src/www/ui/api/Controllers/GroupController.php

```diff
@@ -92,4 +94,59 @@ public function createGroup($request, $response, $args)
     }
     return $response->withJson($returnVal->getArray(), $returnVal->getCode());
   }
+
+  /**
+   * Get users with their roles from a given group
+   *
+   * @param ServerRequestInterface $request
+   * @param ResponseHelper $response
+   * @param array $args
+   * @return ResponseHelper
+   */
+  public function getGroupMembers($request, $response, $args)
+  {
+    $userId = $this->restHelper->getUserId();
+    /** @var UserDao */
+    $userDao = $this->restHelper->getUserDao();
+    $groupMap = $userDao->getAdminGroupMap($userId, $_SESSION[Auth::USER_LEVEL]);
+    $res = null;
+
+    if (empty($groupMap)) {
+      $res = new Info(400, "You have no permission to manage any group.", InfoType::ERROR);
+    } else {
+
+      /**
+       * Get the group id from the params *
+       **/
+      $groupId = intval($args['id']);
+
+      /**
+       * The query to get the list of users with corresponding roles from the group. *
+       **/
+
+      $dbManager = $this->dbHelper->getDbManager();
+
+      $stmt = __METHOD__ . "getUsersWithGroup";
+      $dbManager->prepare($stmt, "select user_pk, user_name, user_status, user_desc, group_user_member_pk, group_perm
+         FROM users LEFT JOIN group_user_member gum ON gum.user_fk=users.user_pk AND gum.group_fk=$1
+         ORDER BY user_name");
```

Comment:
If using previous suggestion, the query can also be simplified to:
```suggestion
      $dbManager->prepare($stmt, "SELECT user_pk, group_perm
         FROM users LEFT JOIN group_user_member gum ON gum.user_fk=users.user_pk AND gum.group_fk=$1
         ORDER BY user_pk");
```

Reply:
And use the `user_fk` instead of `user_pk`.

Reply:
Sir what if I need further details about each user like his username and description.
Because by using that query I can only get the userId.

Reply:
You only need userid, other pieces of information can be fetched from DB as needed. And for the user details in JSON, please check the comment https://github.com/fossology/fossology/pull/2251#discussion_r919991482

It suggests using `$this->dbHelper->getUsers()` to get those information and it just needs userid to fetch the records.

Reply:
sir I would prefer  using that method, I have tried it multiple times, but it keeps on failing popping up this error "_**Expected parameter of type '\Fossology\UI\Api\Models\User', '\Fossology\UI\Api\Models\User[]' provided**_"
I'm going to show you more about this in the upcoming UI meeting.
![Selection_013](https://user-images.githubusercontent.com/66276301/179417350-9b87bdeb-eb4f-4366-98cd-fac1a7c844ea.png)
.

Reply:
i don't know why it's regarding `$user` as an array.